sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py

```diff
@@ -276,7 +284,16 @@ def get_service_properties(self, **kwargs):
         """
         timeout = kwargs.pop('timeout', None)
         try:
-            return self._client.service.get_properties(timeout=timeout, **kwargs)
+            service_props = self._client.service.get_properties(timeout=timeout, **kwargs)
```

Comment:
can we put this block of code into _deserialize? so it could be shared by both async and sync? or we can create an internal class *Properties class and cast from StorageServiceProperties to *Properties

Reply:
I believe we should return a dict (@johanste @annatisch ), but yes sharing the deserialization logic makes sense.

Reply:
It's totally ok for me to have either an dictionary or an object as return type! Not sure if the customer would mind if they have inconsistent experience for get_service_properties, get_container_properties and get_blob_properties. Currently get_container_properties also return a ContainerProperties object(internal type). Let's see how other teammates feel ^_^

Reply:
Discussed with @annatisch to return a dict. Pushed a couple of commits to share the deserialization ;)